docs/FAQ.md

```diff
@@ -0,0 +1,83 @@
+# Registry past-and-present cheat sheet
+
+> Everything You Always Wanted to Know about the Registry
+
+## What's a Registry?
+
+It's a server-side storage and delivery software that let you publish and exchange docker images.
+
+Whenever you do a `docker pull` or `docker push`, your engine communicates with a registry.
+
+## So, does the Docker Hub contain a registry?
+
+Yes.
+
+That "Hub registry" is operated by Docker.
+
+It is free to use for public content, and offers paid-for private storage as well.
+
+Support for the hub is handled directly by Docker, over here: https://support.docker.com
+
+## What if I don't want to rely on the Docker Hub?
+
+Then you should run and operate your own registry. 
+
+While this solution will require more work from you (like running and operating any kind of service does), it will give you more control over your images, and more configurability.
+
+## Are there different registries implementations available?
+
+Yes, there exist at least two different.
+
+First, the "old, legacy registry":
+
+ * implemented in python
+ * source code is here: https://github.com/docker/docker-registry
+ * supports the v1 protocol
+ * is currently in maintenance mode: only critical security fixes are being worked on
+
+Second, the "new registry":
+
+ * implemented in go
+ * source code is here: https://github.com/docker/distribution
+ * implements a new, "v2" protocol
+ * is were most of the development is happening, and the right place for new and upcoming ideas 
+
+
```

Comment:
I think that's out of scope of this document.
Also... sure they can implement their own - it's open-source.